src/_macosx.m

```diff
@@ -178,6 +178,101 @@ static int wait_for_stdin(void)
     return 1;
 }
 
+static PyObject* set_icon(PyObject* ignored, PyObject* path)
+{
+    if (!path)
+    {
+        PyErr_SetString(PyExc_RuntimeError, "No path passed");
+        return NULL;
+    }
+
+    const char* path_string;
+    PyObject* unicode_path = PyObject_Str(path);
+    if (!unicode_path)
+    {
+        return NULL;
+    }
+
+    // API indicates Python owns path_string; we don't need to free it
+    path_string = PyUnicode_AsUTF8(unicode_path);
+    Py_DECREF(unicode_path);
```

Comment:
that looks like path_string may be dangling? (because unicode_path will be gc'd and so path_string will be free'd)

Reply:
Ha. Good call. Weird that this worked anyways.

Reply:
See latest push.